account_invoice_comment_template/views/res_partner_views.xml

```diff
@@ -1,14 +1,13 @@
 <odoo>
 
-    <record id="view_partner_form" model="ir.ui.view">
-        <field name="name">res.partner.form</field>
+    <record id="view_partner_property_form" model="ir.ui.view">
         <field name="model">res.partner</field>
-        <field name="inherit_id" ref="base.view_partner_form"/>
+        <field name="inherit_id" ref="account.view_partner_property_form"/>
         <field name="arch" type="xml">
-            <xpath expr="//page[@name='sales_purchases']//field[@name='user_id']"
-                   position="after">
-                <field name="comment_template_id"
-                       attrs="{'invisible': ['|', ('customer', '=', False), ('is_company', '=', False)]}"/>
+            <xpath expr="//page[@name='accounting']/group[1]" position="inside">
```

Comment:
Please don't move this because it's on purpose on "Sales & Purchases" tab (it's not for vendor bills for example).

Reply:
The invoicing page is automatically hidden when the partner is not the commercial_partner_id, so we benefict from that.. and also the note is only for invoices so doesn't it make sense to have it in the invoicing page?

Reply:
I see one problem with this: if a contact have another language, then you can't put other template in their language. We should make it translatable then for mitigating this problem. The comments are also for sales orders (see `sale_order_comment_template` module).

Reply:
Regarding language: the template is translatable and it's selected based on the partner language when the note is generated.

Regarding `sale_order_comment_template`.. Ok, I see what you mean now. I thought this field was only used in `account_invoice_comment_template` because it's defined here instead of being defined in `base_comment_template`.

So, that field value affects both invoices and sale orders (if both modules are installed).

To be honest that doesn't make any sense to me. I would prefer to add an `model_id` field in the `base.comment.template` model and then have the partner has multiple fields: one for invoices and other for sale orders, based on the model_id field of the comment template. Then, `sale_order_comment_template` wouldn't have to depend on `account_invoice_comment_template`..

Also right now it's impossible to set both `before_lines` and `after_lines` comment templates for a partner. You can only choose one of them.

Reply:
I think that definition is double right now: in `base_comment_template` and here. Please check.

Reply:
No, `base_comment_template` doesn't define any fields in `res.partner`
https://github.com/OCA/account-invoice-reporting/tree/12.0/base_comment_template/models

Reply:
Then it should be defined there, not here. Can you please move it?

Reply:
Done